### ISSUE TITLE
Youtube comments can't be posted under videos after the most recent update(s)

### ISSUE THREAD

**Issue by InvisibleManX** [MrAlex94/Waterfox#2180]
### What happened?

I'm using the most recent Waterfox G3.2.4.1 and when I try to leave a comment under a youtube video it seems to be posted, but once I refresh/reload the page, my comment isn't there at all! So it's not actually posted. 

I tested it multiple times and my friends tested it. 

I can leave the comments just fine from Chrome, but Waterfox doesn't allow me to leave comments at all.... 

It could be related to the browser's identification which Google banned or something like that. I don't know for sure. 

But the weirdest part is that it prevents me to leave comments only under other channels' videos. But, under the videos on my own channel, my comments do get posted just fine. It's really weird!

Please, fix this issue. Youtube is unusable now! :(

### Reproducible?

- [x] I have tested this on Firefox.

### Version

G3.𝑋 (Current)

### What platform are you seeing the problem on?

Windows

**Comment by mkruselj** [comment]
This is preposterous.

Same thing happening in WF Classic.

**Comment by Wasmachineman-NL** [comment]
wow and I thought I got commentbanned from YT, thankfully it's just a bug in WFC.

**Comment by hawkeye116477** [comment]
I wouldn't be surprised as if it turned out that it was Google's fault. Maybe they broke their browser detection code and it forbids Waterfox now...

See also https://forum.palemoon.org/viewtopic.php?f=70&t=27007&p=218610&hilit=youtube+comment#p218610.

**Comment by InvisibleManX** [comment]
@hawkeye116477 

Here is something interesting from someone in comments on this page: [https://support.google.com/youtube/thread/116351041/recent-reports-of-missing-comments](https://support.google.com/youtube/thread/116351041/recent-reports-of-missing-comments)

> I cannot confirm this with certainty yet, but it appears that changing my Firefox settings¹ resolved it for me. I had a comment disappearance rate of around ~75% (i.e. most) for the past months, and now it appears they all stick. I don't dare test this much however, in case the reason lies elsewhere and I jeopardize that by commenting more again.

> ¹: Settings like network.http.referer.XOriginPolicy and possibly RFP related settings, that might have causes a mismatch between my IP's location and the client's reported timezone (UTC+0).

**Comment by InvisibleManX** [comment]
@hawkeye116477 
@MrAlex94 

Here is something interesting from someone in comments on this page: [https://support.google.com/youtube/thread/116351041/recent-reports-of-missing-comments](https://support.google.com/youtube/thread/116351041/recent-reports-of-missing-comments)

> I cannot confirm this with certainty yet, but it appears that changing my Firefox settings¹ resolved it for me. I had a comment disappearance rate of around ~75% (i.e. most) for the past months, and now it appears they all stick. I don't dare test this much however, in case the reason lies elsewhere and I jeopardize that by commenting more again.

> ¹: Settings like `network.http.referer.XOriginPolicy` and possibly RFP related settings, that might have causes a mismatch between my IP's location and the client's reported timezone (UTC+0).

So possibly it's not entirely Google's or Youtube's fault, but something that should be set-up different in Firefox based browsers now, to use youtube without that issue.

**Comment by asbestinuS** [comment]
Is there any knowledge on if and when this will be fixed? I noticed it a few months ago already (on Waterfox classic) and thought my comments went directly to spam, but that wasn't the case. It's clearly a Waterfox issue beause on any other broweser, my comments would stay.
Any update on that, please?

**Comment by hawkeye116477** [comment]
Official statement is that for Classic you will only get security updates. In case of new features they could be only provided by external contributors through Pull Requests, but in that case seems that no one knows what's the cause exactly.
Same problem is on [Pale Moon](https://forum.palemoon.org/viewtopic.php?f=70&t=27007&p=219672&hilit=youtube#p219672).

**Comment by Sollace** [comment]
fyi I was able to fix youtube comment posting on 90% of channels by changing my userAgent to reflect the one used by FF91.

i.e. like so in about:config
![image](https://user-images.githubusercontent.com/6429283/135851870-168b5121-3459-4c03-8671-e2475cb6a5fb.png)

I believe the problem is google has begun blocking requests from older browsers which, since it identifies as Firefox 78 right now, means YouTube will also block anyone using Waterfox.

**Comment by InvisibleManX** [comment]
@hawkeye116477 
@MrAlex94 

So as the recent comment here says, it is possible to fix it if an individual user agent override would be added for youtube. 

The simple editing in about:config didn't work for me even after the browser restart, but trying to do it with some random addons that allow to switch user agent string worked, and youtube comments can only be posted if waterfox is identified as newer firefox verisons.

**Comment by hawkeye116477** [comment]
Well, in case of Classic, newer user-agent is bad idea rather, cuz switching that can cause that YT will switch to Polymer v2, so webcomponents will be required. I didn't tested how it's on G3.

**Comment by InvisibleManX** [comment]
@hawkeye116477 
Seems to work fine on G3 ! 
I cleared cookies, restarted a browser, and it finally worked with just that `general.useragent.override` string,
Google or youtube, or other sites don't see to be broken. 

But I still would prefer a way so the site can be only specifically for youtube. Just in case. So would be nice to have that in future waterfox upgrades. 

The thing is, I tried this kind of string `general.useragent.override.youtube.com` and apparently those don't work anymore (but had to work in the past). So only the global setting works.

**Comment by InvisibleManX** [comment]
Though it seems that it worked only after I also added `navigator.userAgent` parameter with the same string. (I read about adding it on some site). I did it, because first time I tired that general.useragent.override alone it didn't seem to work. 

Maybe just a coincidence. 

But still I notice that one some videos the messages still don't go through, or when replying to some people. 

Maybe there's also some blacklist for specific words in messages on youtube. I don't know.

**Comment by Sollace** [comment]
@InvisibleManX Yeah, unfortunately that's the other 10% I mentioned still doesn't work. Switching to normal firefox to post the same comments goes through, so I don't believe it's anything to do with the comment itself. Rather I think youtube/google are doing some browser fingerprinting and there's some feature missing in waterfox that they're looking for to determine legitimacy.

**Comment by MrAlex94** [comment]
I have tested out some comments on YT with G4 - appears to be working as expected now. If you guys could test and let me know if you still have the issue? I don't comment all too much on YT.

**Comment by InvisibleManX** [comment]
> 
> I have tested out some comments on YT with G4 - appears to be working as expected now. If you guys could test and let me know if you still have the issue? I don't comment all too much on YT.

I figured out the cause of the problem with the help of people who replied here before. 

So the whole reason was an outdated Firefox ESR 78 (so G3 was affected, but it was possible to fix it by using the user agent override for Firefox ESR 91). 

Now, G4 is based on ESR 91, so it will work, assuming it would use that same user agent string to identify itself, as Firefox. 

Because I tested everything on Firefox ESR 91 and everything works well on youtube. 

I couldn't test it on G4 for the reason of being denied the right to use it on the account of me not having SSE 4.2 in my CPU....  XD
